d2bs/kolbot/libs/modules/Town.js

```diff
@@ -352,6 +352,26 @@
 		return true;
 	};
 
+	// Purchases and drinks potions in town.
+	Town.stackPotions = function(type, amount) {
+        const npc = Town.initNPC("Shop", "buyPotions");
+        for(var i = 0; i < amount; i++){
+			const npcPotion = Town.getPotion(npc, type);
+			if (npcPotion) {
+				npcPotion.buy(false);
+				const potion = me.getItem(npcPotion.classid);
+				if (potion) {
+					do {
+						if ((potion.mode === 0 && potion.location === 3) || potion.mode === 2) {
```

Comment:
i know, because i written allot for kotlon and diablo II in general that 3 is inventory, however lets make the code document itself.

Instead of 3, use sdk.storage.Inventory.

Also; Since the pot needs to be mode 0 and in inventory, or mode 2.. This might cause an issue if you have an pot in stash/cube, which you try to interact with, which you cant as your not interacting with the stash/cube?

Might need to be changed to in inventory, or belt. And mode 2, or 0?

Reply:
I see now mode 2 is in belt.. Still, this is one of those reasons to use sdk's aliases,

```javascript
sdk.itemmode.inBelt;  // Documents it self =)
```

Reply:
yes plz try to avoid hard coded values, a lot of them can be replaced by `sdk.foo.bar`

Reply:
also use `me.getItems(...)` to retrieve an array, instead of a magic over complex C struct with dumb `getNext`